crates/gitbutler-diff/src/write.rs

```diff
@@ -62,7 +63,13 @@ where
             && hunks[0].diff_lines.contains_str(b"Subproject commit");
 
         // if file exists
-        if full_path.exists() {
+        if full_path.exists() || allow_new_file {
+            if hunks.len() == 1 && hunks[0].change_type == crate::ChangeType::Deleted {
+                // File was created but now that hunk is being discarded with an inversed hunk
+                builder.remove(rel_path);
+                fs::remove_file(full_path.clone())?;
```

Comment:
Can we be sure here that the file actually exists on disk? As far as I am concerned, hunks can be arbitrary and `allow_new_file` could be `true`.